File: 2022/01.rb
require "pathname"

path = Pathname.new __FILE__
inputPath =
  Pathname.new File.dirname(path) + "/input/" +
                 File.basename(path).split(".")[0].to_i.to_s + ".input"
input = File.read(inputPath)

elves =
  input
    .split("\n\n")
    .map { |string| string.split("\n").map(&:to_i).sum }
    .sort
    .reverse
puts elves[0]
puts elves[0...3].sum

File: 2022/01.py
import os
import sys

path = os.path.abspath(sys.argv[0])
input_path = (
    os.path.dirname(path)
    + "/input/"
    + str(int(os.path.basename(path).split(".")[0]))
    + ".input"
)

with open(input_path) as file:
    input = file.read().strip()
    elves = list(
        map(
            lambda e: sum(list(map(lambda s: int(s), e.split("\n")))),
            input.split("\n\n"),
        )
    )
    elves.sort(reverse=True)
    print(elves[0])
    print(sum(elves[:3]))
